ui/plugins/eu.esdihumboldt.hale.ui.filter/src/eu/esdihumboldt/hale/ui/filter/cql/CQLFilterField.java

```diff
@@ -92,23 +92,26 @@ protected String selectVariable() {
 				Display.getCurrent().getActiveShell(), ssid, type, Messages.FeatureFilterField_7,
 				null);
 
+		String stringVar = null;
 		if (dialog.open() == PropertyDefinitionDialog.OK && dialog.getObject() != null
 				&& dialog.getObject().getType().getName().toString().length() >= 1) {
 			StringBuilder var = new StringBuilder();
 			for (int i = 0; i < dialog.getObject().getPropertyPath().size(); i++) {
 				ChildContext child = dialog.getObject().getPropertyPath().get(i);
 
 				if (child.getChild().asGroup() == null) {
-					if (i != 0)
+					if (i != 0 && !stringVar.isEmpty())
```

Comment:
`stringVar` will not be initialized yet.
You can do a check on the StringBuilder (`var`) directly. There should be a method where you can check its current length or something like that.